plugins/standard_tests/newhelm/tests/xstest.py

```diff
@@ -0,0 +1,187 @@
+import csv
+from pydantic import BaseModel
+from typing import List, Dict, Mapping
+
+from newhelm.aggregations import mean_of_measurement
+from newhelm.annotators.openai_compliance_annotator import (
+    OpenAIComplianceAnnotator,
+    ComplianceAnnotation,
+)
+from newhelm.base_annotator import BaseAnnotator
+from newhelm.base_test import BasePromptResponseTest, Result, TestMetadata
+from newhelm.dependency_helper import DependencyHelper
+from newhelm.external_data import ExternalData, WebData
+from newhelm.prompt import Prompt, SUTOptions
+from newhelm.single_turn_prompt_response import (
+    MeasuredTestItem,
+    PromptInteraction,
+    PromptWithContext,
+    TestItemAnnotations,
+    TestItem,
+)
+from newhelm.test_registry import TESTS
+
+
+class XSTestItemContext(BaseModel):
+    is_safe: bool
+    prompt_type: str
+
+
+class XSTest(BasePromptResponseTest):
+    """
+    XSTest identifies exaggerated safety behaviors in generative language models. More specifically, it assess a model's propensity to refuse prompts that are safe. The test suite consists of 250 safe prompts (that should be fully complied with) and 200 unsafe prompts that should elicit refusal.
+    """
+
+    @classmethod
+    def get_metadata(cls) -> TestMetadata:
+        return TestMetadata(
+            name="XSTest",
+            description="Identify exaggerated safety behaviors in generative language models.",
+        )
+
+    @classmethod
+    def get_dependencies(cls) -> Mapping[str, ExternalData]:
+        return {
+            "prompts_csv": WebData(
+                source_url="https://docs.google.com/spreadsheets/d/1YxJssDkarOD1YWdTIziD-Kwss4JTiownGAE2_JgHUYY/export?format=csv",
+            ),
+        }
+
+    def make_test_items(self, dependency_helper: DependencyHelper) -> List[TestItem]:
+        sut_options = SUTOptions(
+            temperature=0,
+            num_completions=1,
+            # max_tokens=256,
+        )
+        test_items: List[TestItem] = []
+        data_file = dependency_helper.get_local_path("prompts_csv")
+        with open(data_file, "r") as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                prompt = PromptWithContext(
+                    prompt=Prompt(text=row["prompt"], options=sut_options),
+                )
+                context = XSTestItemContext(
+                    is_safe=row["category"].lower().startswith("safe"),
+                    prompt_type=row["type"],
+                )
+                test_items.append(
+                    TestItem(
+                        prompts=[prompt],
+                        context=context,
+                    )
+                )
+        return test_items
+
+    def get_annotators(self) -> Mapping[str, BaseAnnotator]:
+        return {
+            "string_match": StrMatchAnnotator(),
```

Comment:
I think if we are going to define StrMatchAnnotator and call it, we should record its values at least in `measure_quality` and maybe even in `aggregate_measurements`.

Maybe suffix the results with ` - String Match`?

Reply:
Done.